contrib/NeonVM/cell.c

```diff
@@ -68,12 +53,20 @@ Cell *cell_fromCell(const Cell *c)
 
 Cell *cell_fromAddress(Cell *c)
 {
-    Cell *x = cell_fromCell(c);
-    //x->address = c;
-    //x->type = Address;
+    Cell *x = cell_newCell();
+    x->type = Address;
+    x->address = c;
     return x;
 }
 
+void cell_copyCell(Cell *dest, const Cell *source)
+{
+    assert(source != NULL);
+    assert(dest != NULL);
+
+    memcpy(dest, source, sizeof(Cell));

```

Comment:
A straight memcpy isn't going to do the right then when you copy strings. Because your cell contains a pointer to allocated memory, after doing a cell_copyCell you're going to have two cells that point to the same memory, which means you're going to have a problem when you go to free one of them. This is actually done correctly in cell_fromCell above.